### PR TITLE
Fix Add using under chained conditional access expressions.

### LINKED ISSUE #3818: Quick action unavailable when chaining and using null conditional.
There is no quick action (Ctr + .) available when using the null conditional and chaining operations together.

Self-contained example below:

```CSharp

namespace Sample
{
    class Program
    {
        static void Main(string[] args)
        {
            string myString = "Sample";
            var result = myString.StringExtension().Substring(0); // Quick action available
            var other = myString?.StringExtension().Substring(0); // No quick action available
        }
    }
}

namespace Sample.Extensions
{
    public static class StringExtensions
    {
        public static string StringExtension(this string s)
        {
            return "Ok";
        }
    }
}

```

image

### PR DESCRIPTION
**Pull request by balajikris**
Handle ConditionalAccessExpression cases properly in `GetExpressionOfMemberAccessExpression` by using `GetParentCondtionalAccessExpression` helper to find a ConditionalAccessExpression if one exists in a parent hierarchy. This change is C# only. The VB implementation of this method
correctly uses `GetCorrespondingConditionalAccessExpression` in this case.

Fixes #3818.

### CODE REVIEW

**Comment by balajikris** [comment]
@brettfo @CyrusNajmabadi @davkean @dpoeschl @DustinCampbell @jmarolf @rchande @Pilchie @jasonmalinowski : Please review.

File: src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs [hunk 1]
```diff
@@ -186,5 +186,99 @@ public void TestAddUsingForAddExtentionMethod11()
...
+        {
+            var initialText =
+  @"<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly"" CommonReferences=""true"">
+        <Document FilePath = ""Program"">
+public class C
+{
+    public T F&lt;T&gt;(T x)
+    {
+        return F(new C())?.F(new C())?[|.Extn()|];
```
Comment (dpoeschl) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs]: Are there any other interesting test cases? I see you've got ```A?.TEST.C``` and ```A?.B?.TEST```. Is ```TEST?.A``` or ```A?B.TEST?.C``` or anything else interesting? I'm certainly not saying these tests are insufficient, I just know this is a complex area and we've (I've) underestimated it before. :smile:

**Comment by jmarolf** [comment]
:+1:

File: src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs (hunk 1 above)
Comment (balajikris) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs]: @dpoeschl I understand (from previous experiences as well :smile: ) . the change here is simple - just reuse a helper which has been well tested. In theory we should be fine. That said, I find your `A?B.TEST?.C` interesting.. Let me add a test for that case :wink:

**Comment by dpoeschl** [comment]
:+1:

File: src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs [hunk 2]
```diff
@@ -555,20 +555,9 @@ public void GetNameAndArityOfSimpleName(SyntaxNode node, out string name, out in
...
-                }
-                if (node.IsParentKind(SyntaxKind.InvocationExpression) &&
-                    node.Parent.IsParentKind(SyntaxKind.ConditionalAccessExpression))
-                {
-                    return GetExpressionOfConditionalMemberAccessExpression(node.Parent.Parent);
-                }
-            }
-
-            return (node as MemberAccessExpressionSyntax)?.Expression;
+            return node.IsKind(SyntaxKind.MemberBindingExpression) ?
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs]: ? and : on newlines.

Comment (dpoeschl) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs]: (seconded)

Comment (DustinCampbell) [inline src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs]: and thirded. This is a stylistic abomination. :smile:

File: src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs (hunk 1 above)
Comment (balajikris) [inline src/EditorFeatures/CSharpTest/Diagnostics/AddUsing/AddUsingTests_ExtensionMethods.cs]: Done.

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**